.github/workflows/build-docker-image.yml

```diff
@@ -74,26 +112,127 @@ jobs:
     runs-on: windows-2022
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-go@v3
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
         with:
-          go-version-file: 'go.mod'
-      - run: go version
+          images: ${{ env.IMAGE_REPOSITORY }}
+          tags: |
+            type=sha,format=long
+          flavor: |
+            suffix=-windows.${{ matrix.os-version }}
+
+      - name: Login to Docker Hub
```

Comment:
We probably need to think about this feature twice: because it runs in a fork, the secrets are not available at the PR level.

Could you skip the docker login/push steps while in PRs?

Reply:
Yes, I'll try to add a check/flag during the next few days.

Reply:
I configured the jobs depending on the Docker Hub to only run on the default branch (main).

Reply:
... oh wait. The 'build' job needs a special treatment.

Reply:
Would it make sense to add a "check" job to be required for pull requests, with the "check" depending on the "test-*" jobs? I'm going to change the "build" dependencies to be reduced to the "test-*" jobs, but I'm open to change the jobs and their structure like you prefer.